### DISCUSSION TITLE
Relative x,y animation is needed

### DISCUSSION THREAD

**Discussion by dmytroreutov** [patrykandpatrick/vico#877] [Ideas]
Hi! 

I'm trying to implement "real-time" line-chart based on `com.patrykandpatrick.vico.views.cartesian.CartesianChartView` for Views framework. I feed data from reactive source to chart view. So the new values are added to existing dataset frequently (several times per second).

Existing animation looks off when adding new data points through `CartesianChartModelProducer`, because it simply jumps from 0 to y value.

So I disabled it via `CartesianChartView#setAnimationDuration(0L)`. 

However, relative (x,y) animation is needed for this purpose, to draw line connecting old (x0,y0) and new (x1,y1) points.

I don't think I can force the source to emit elements smoothly so they can be put into dataset as is. Instead I can throttle elements by fixed timeout and put them into dataset expecting animation connect them with previous based on coordinates.

**Comment by Gowsky** [comment]
Hello @dmytroreutov. That's something we're looking to add in the near future. It can be partially solved with a solution from #885.

**Reply by dmytroreutov** [reply]
Thank you for your response. 
I've created custom `LineDrawingModelInterpolator` with tweaked `updateTransformationMap` :

```
                newDrawingModel?.getOrNull(index)?.forEach { (x, drawingInfo) ->
                    val resultOldDrawInfo = map[x]?.old ?: map[x - 1]?.new
                    map[x] = TransformationModel(resultOldDrawInfo, drawingInfo)
                }
```

So for now it retains y position from previous point. 

Unfortunately, there is no room for customisation of x animation (to apply same fraction behaviour)

**Reply by Gowsky** [reply]
>Unfortunately, there is no room for customisation of x animation (to apply same fraction behaviour)

That's why I said we're looking into adding this kind of animation soon and that a custom solution can partially solve the problem.
